imap_processing/spice/geometry.py

```diff
@@ -87,3 +90,148 @@ def imap_state(
+
+    Parameters
+    ----------
+    start_met : float
+        Provides the start time in Mission Elapsed Time (MET).
+    end_met : float
+        Provides the end time in MET. If not provided, default to one day
+        from start time.
+
+    Returns
+    -------
+    spin_df : dict
+        Spin data. May need to save data to CSV file.
+    """
+    if end_met is None:
+        # end_time is one day after start_time
+        end_met = start_met + 86400
+
+    # Create spin start second data of 15 seconds increment
+    spin_start_sec = np.arange(start_met, end_met, 15)
+
+    spin_dict = {
+        "spin_number": np.arange(spin_start_sec.size, dtype=np.uint32),
+        "spin_start_sec": spin_start_sec,
+        "spin_start_subsec": np.full(spin_start_sec.size, 0, dtype=np.uint32),
+        "spin_period_sec": np.full(spin_start_sec.size, 15.0, dtype=np.float32),
+        "spin_period_valid": np.ones(spin_start_sec.size, dtype=np.uint8),
+        "spin_phas_valid": np.ones(spin_start_sec.size, dtype=np.uint8),
+        "spin_period_source": np.zeros(spin_start_sec.size, dtype=np.uint8),
+        "thruster_firing": np.zeros(spin_start_sec.size, dtype=np.uint8),
+    }
+
+    # Convert spin_start_sec to datetime to set repointing times flags
+    spin_start_dates = met_to_j2000ns(spin_start_sec)
+    spin_start_dates = cdflib.cdfepoch.to_datetime(spin_start_dates)
+
+    # Convert DatetimeIndex to Series for using .dt accessor
+    spin_start_dates_series = pd.Series(spin_start_dates)
+
+    # Find index of all timestamps that fall within 10 minutes after midnight
+    repointing_times = spin_start_dates_series[
+        (spin_start_dates_series.dt.time >= pd.Timestamp("00:00:00").time())
+        & (spin_start_dates_series.dt.time < pd.Timestamp("00:10:00").time())
+    ]
+
+    repointing_times_index = repointing_times.index
+
+    # Use the repointing times to set thruster firing flag and spin period valid
+    spin_dict["thruster_firing"][repointing_times_index] = 1
+    spin_dict["spin_period_valid"][repointing_times_index] = 0
+    spin_dict["spin_phas_valid"][repointing_times_index] = 0
+
+    return spin_dict
+
+
+def get_spin_data(start_met: float, end_met: Optional[float] = None) -> dict:
+    """
+    Get spin data for a given time range.
+
+    This function queries spin data for the input date range. Spin
+    table contains the following fields:
+        (
+            spin_number,
+            spin_start_sec,
+            spin_start_subsec,
+            spin_period_sec,
+            spin_period_valid,
+            spin_phas_valid,
+            spin_period_source,
+            thruster_firing
+        )
+
+    Parameters
+    ----------
+    start_met : float
+        Provide the start time in Mission Elapsed Time (MET).
+    end_met : float
+        Provide the end time in MET. If not provided, default to one day
+        from start time.
+
+    Returns
+    -------
+    spin_data : dict
```

Comment:
I had that originally as we discussed but then I was thinking about others who may want to use this function but don't want their data in pandas data form. Because of that, `dict` felt generic enough. But I can change it back to  return `pd.DataFrame` like we originally thought.

Reply:
I changed to return `pd.DataFrame` and if we decide to change down the road, we can do that.